### PR TITLE
Update syntax for installing from git repos to latest recommended

### LINKED ISSUE #11: Re-enable sunpy when we can
Their dependencies got too complicated. I don't think simple install command here can handle that and I don't feel like writing a bunch of exceptions just for them. They should test astropy RC themselves.

Disabled in #10

### PR DESCRIPTION
**Pull request by astrofrog**
Fixes #11 (the sunpy build)

There are however now genuine test failures in the sunpy build

### CODE REVIEW

**Comment by astrofrog** [comment]
@pllim - I think it does make sense to run in PRs, so I think we should permanently add that setting.

**Comment by astrofrog** [comment]
I think the test failures might be related to the fact that there are some pytest options specified in sunpy's setup.cfg, but we'd have to repeat these here as command-line arguments. I wonder if there's a way to avoid that though as that would be a pain to keep in sync. Will ping the sunpy folk.

File: .github/workflows/integration_testing.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ name: astropy_rc_basic
 
 on:
   workflow_dispatch:
+  pull_request:
```
Comment (pllim) [inline .github/workflows/integration_testing.yml:5]: If there is no security concern, I guess it is okay to leave this in? This is a low-traffic repo anyway.

Comment (astrofrog) [reply .github/workflows/integration_testing.yml:5]: I don't see any potential security concerns with this. And any PR is likely to change the config in a way we want to make sure it isn't breaking I think.

File: tox.ini [hunk 2]
```diff
@@ -24,17 +24,17 @@ pip_pre = true
 # side effects and make sure all test suites pass with all packages
 deps =
     astropy[all,test]
-    asdf_astropy,all: git+https://github.com/astropy/asdf-astropy.git#egg=asdf_astropy[test]
-    astropy_healpix,all: git+https://github.com/astropy/astropy-healpix.git#egg=astropy_healpix[test]
-    astroquery,all: git+https://github.com/astropy/astroquery.git#egg=astroquery[test,all]
+    asdf_astropy,all: asdf_astropy[test] @ git+https://github.com/astropy/asdf-astropy.git
+    astropy_healpix,all: astropy_healpix[test] @ git+https://github.com/astropy/astropy-healpix.git
+    astroquery,all: astroquery[test,all] @ git+https://github.com/astropy/astroquery.git
     ccdproc,all: psutil
```
Comment (pllim) [inline tox.ini:30]: Oops did I use the wrong syntax to pull in dependencies? My bad!

Comment (astrofrog) [reply tox.ini:30]: I think this original syntax did work at some point but I think it's not the latest and maybe doesn't work with new features like what sunpy uses. In any case, this should now be the preferred syntax.

**Review by pllim — APPROVED** [review]
The diff looks good. Should we investigate the actual sunpy failure separately?

**Comment by astrofrog** [comment]
Yes sounds good

**Pull request merged**